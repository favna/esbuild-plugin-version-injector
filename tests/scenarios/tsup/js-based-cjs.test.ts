import tsup, { type Options } from 'tsup';
import type { PluginOptions } from '../../../dist';
import { assertFileContent, buildAbsolutePath, createTsupConfig } from '../../common/util';

function createConfig(pluginOptions?: PluginOptions): Options {
  return createTsupConfig(
    {
      format: ['cjs'],
      entry: [buildAbsolutePath('../fixtures/build-in/javascript.cjs')]
    },
    pluginOptions
  );
}

describe('JS based CJS', () => {
  test('GIVEN no plugin options THEN injects version', async () => {
    const config = createConfig();

    await tsup.build(config);

    await assertFileContent('../fixtures/build-out/tsup/javascript.cjs');
  });

  test('GIVEN versionOrCurrentDate = "current-date" THEN injects current date', async () => {
    const config = createConfig({
      versionOrCurrentDate: 'current-date'
    });

    await tsup.build(config);

    await assertFileContent('../fixtures/build-out/tsup/javascript.cjs');
  });
});
