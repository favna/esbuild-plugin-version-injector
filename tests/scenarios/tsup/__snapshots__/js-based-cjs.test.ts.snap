// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JS based CJS > GIVEN no plugin options THEN injects version 1`] = `
"'use strict';

var __getOwnPropNames = Object.getOwnPropertyNames;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};

// tests/fixtures/build-in/javascript.cjs
var require_javascript = __commonJS({
  "tests/fixtures/build-in/javascript.cjs"(exports, module) {
    module.exports = {
      version: "1.2.1"
    };
  }
});
var javascript = require_javascript();

module.exports = javascript;
"
`;

exports[`JS based CJS > GIVEN versionOrCurrentDate = "current-date" THEN injects current date 1`] = `
"'use strict';

var __getOwnPropNames = Object.getOwnPropertyNames;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};

// tests/fixtures/build-in/javascript.cjs
var require_javascript = __commonJS({
  "tests/fixtures/build-in/javascript.cjs"(exports, module) {
    module.exports = {
      version: "2022-02-01T14:30:30.000Z"
    };
  }
});
var javascript = require_javascript();

module.exports = javascript;
"
`;
